app/gui/qt/mainwindow.cpp

```diff
@@ -1323,11 +1326,39 @@ void MainWindow::startupError(QString msg) {
     QSpacerItem* hSpacer = new QSpacerItem(200, 0, QSizePolicy::Minimum, QSizePolicy::Expanding);
     layout->addItem(hSpacer, layout->rowCount(), 0, 1, layout->columnCount());
     box->exec();
+
+    // attempt to fully close all server processes
+    stopServer();
+
     std::cout << "[GUI] - Aborting. Sorry about this." << std::endl;
     QApplication::exit(-1);
     exit(EXIT_FAILURE);
 }
 
+void MainWindow::stopServer() {
+    // Attempt to stop the server
+    if (serverProcess) {
+        if (serverProcess->state() == QProcess::NotRunning) {
+            std::cout << "[GUI] - server process is not running." << std::endl;
+        } else if (!oscSender) {
+            std::cout << "[GUI] - OSC sender is not initialised." << std::endl;
+        } else {
+            std::cout << "[GUI] - attempting to stop the server..." << std::endl;
+            Message msg("/exit");
+            msg.pushStr(guiID.toStdString());
+            sendOSC(msg);
+        }
+    } else {
+      std::cout << "[GUI] - server process hasn't been initialised." << std::endl;
+    }
+    sleep(2);
+    // ensure all child processes are nuked if they didn't die gracefully
+    std::cout << "[GUI] - executing exit script" << std::endl;
```

Comment:
This line of code is now extracted as part of #2418
Also not sure why this PR was not accepted, but if you split it in 2 parts, one configuration of startup timeout, second one is changes in process shutdown.

Overall looks great to me.

/cc @samaaron